internal/state/graph/httproute.go

```diff
@@ -403,6 +417,33 @@ func findAcceptedHostnames(listenerHostname *v1beta1.Hostname, routeHostnames []
 	return result
 }
 
+func routeAllowedByListener(
+	listener *Listener,
+	routeNS,
+	gwNS string,
+	namespaces map[types.NamespacedName]*apiv1.Namespace,
+) bool {
+	if listener.Source.AllowedRoutes != nil {
+		switch *listener.Source.AllowedRoutes.Namespaces.From {
+		case v1beta1.NamespacesFromAll:
+			return true
+		case v1beta1.NamespacesFromSame:
+			return routeNS == gwNS
+		case v1beta1.NamespacesFromSelector:
+			if listener.AllowedRouteLabelSelector == nil {
+				return false
+			}
+
+			ns := namespaces[types.NamespacedName{Name: routeNS}]
+			if listener.AllowedRouteLabelSelector.Matches(labels.Set(ns.Labels)) {
```

Comment:
Is it possible that the route namespace doesn't exist in the namespaces map?

Reply:
Every namespace in the system gets added to the map, though I can add a check here just to be safe.

Reply:
Yeah, let's add a check. I think we want to panic here anyway but with an informative message rather than just panicking when we try to access the labels from the namespace.